survey-sdk/src/test/java/co/nimblehq/survey/serializer/SerializerTest.kt

```diff
@@ -0,0 +1,150 @@
+package co.nimblehq.survey.serializer
+
+import co.nimblehq.survey.sdk.entity.AnswerEntity
+import co.nimblehq.survey.sdk.entity.MetaEntity
+import co.nimblehq.survey.sdk.entity.QuestionEntity
+import co.nimblehq.survey.sdk.entity.SurveyEntity
+import com.squareup.moshi.JsonAdapter
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.kotlin.reflect.KotlinJsonAdapterFactory
+import moe.banana.jsonapi2.ArrayDocument
+import moe.banana.jsonapi2.Document
+import moe.banana.jsonapi2.ObjectDocument
+import moe.banana.jsonapi2.ResourceAdapterFactory
+import org.junit.Assert.assertNotNull
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.junit.MockitoJUnitRunner
+
+@RunWith(MockitoJUnitRunner::class)
+class SerializerTest {
+    lateinit var moshi: Moshi
+
+    @Before
+    fun setup() {
+        val jsonApiAdapterFactory: JsonAdapter.Factory = ResourceAdapterFactory.builder()
+            .add(SurveyEntity::class.java)
+            .add(QuestionEntity::class.java)
+            .add(AnswerEntity::class.java)
+            .build()
+        moshi = Moshi.Builder()
+            .add(jsonApiAdapterFactory) // ...
+            .add(KotlinJsonAdapterFactory())
+            .build()
+    }
+
+    @Test
+    fun `when parsing the right json string as list survey, it should return the SurveyListEntity`() {
+
+        val surveyList: ArrayDocument<SurveyEntity> =
+            moshi.adapter(Document::class.java)
+                .fromJson(JSON_SURVEY_LIST_TEST) as ArrayDocument<SurveyEntity>
+        assertTrue(surveyList.isNotEmpty())
+        //item test
+
+        val firstItem = surveyList.first()
+        //attributes test
+        assertTrue(firstItem.title == "Scarlett Bangkok")
+        assertTrue(firstItem.description == "We'd love ot hear from you!")
+        assertTrue(firstItem.thankEmailAboveThreshold == "Test1")
+        assertTrue(firstItem.thankEmailBelowThreshold == "Test2")
+        assertTrue(firstItem.isActive == true)
+        assertTrue(firstItem.coverImageURL == "https://dhdbhh0jsld0o.cloudfront.net/m/1ea51560991bcb7d00d0_")
+        assertTrue(firstItem.createdAt == "2017-01-23T07:48:12.991Z")
+        assertTrue(firstItem.activeAt == "2015-10-08T07:04:00.000Z")
+        assertTrue(firstItem.surveyType == "Restaurant")
+
+        //relationship test
+        assertNotNull(firstItem.questions)
+        val questionList = firstItem.getQuestionList()
+        assertTrue(questionList?.isNotEmpty() == true)
+        //first data test
+        val firstData = questionList?.first()
+        assertTrue(firstData?.id == "d3afbcf2b1d60af845dc")
+        assertTrue(firstData?.type == "question")
```

Comment:
I think we could use `with` to get rid of repeating `firstData` to get the property fields.

Reply:
@hoangnguyen92dn Agree with this